drivers/video/fb.c

```diff
@@ -682,6 +675,30 @@ static int fb_ioctl(FAR struct file *filep, int cmd, unsigned long arg)
   return ret;
 }
 
+static int     fb_mmap(FAR struct file *filep, FAR void **addr, off_t start,
+                       size_t length)
+{
+  FAR struct inode *inode;
+  FAR struct fb_chardev_s *fb;
+  int ret = -EINVAL;
+
+  /* Get the framebuffer instance */
+
+  DEBUGASSERT(filep != NULL && filep->f_inode != NULL);
+  inode = filep->f_inode;
+  fb    = (FAR struct fb_chardev_s *)inode->i_private;
+
+  /* Return the address corresponding to the start of frame buffer. */
+
+  if (addr)
+    {
+      *addr = fb->fbmem;
```

Comment:
add start to fb->fbmem

Reply:
sorry, don't understand.

this is the original code
```
-      case FIOC_MMAP:  /* Get color plane info */
-        {
-          FAR void **ppv = (FAR void **)((uintptr_t)arg);
-
-          /* Return the address corresponding to the start of frame buffer. */
-
-          DEBUGASSERT(ppv != NULL);
-          *ppv = fb->fbmem;
-          ret = OK;
-        }
-        break;
```
I am not (deliberately) changing any functionality of the existing drivers in this PR

Reply:
Before your change, FIOC_MMAP doesn't pass offset to the callback, so the driver can only return the start of buffer. But with the new mmap callback, the driver can adjust the returned value to consider offset required by caller.

Reply:
Ok, I will add the checks